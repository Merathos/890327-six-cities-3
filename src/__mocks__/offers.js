export default [
  {
    id: `41324`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3909553943508, 4.85309666406198],
    zoom: 10,
    title: `Beautiful & luxurious apartment at great location`,
    previewImg: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 3,
    isBookmarked: true,
    isPremium: false,
    price: 120,
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    hostStatus: true,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `73645`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.369553943508, 4.85309666406198],
    title: `Wood and stone place`,
    previewImg: `img/apartment-02.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 2,
    maxAdults: 3,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    type: `Private room`,
    rating: 4,
    isBookmarked: false,
    isPremium: true,
    price: 80,
    hostName: `Max`,
    hostAvatar: `img/avatar-max.jpg`,
    hostStatus: false,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `23455`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3909553943508, 4.929309666406198],
    title: `Canal View Prinsengracht`,
    previewImg: `img/apartment-03.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 3,
    maxAdults: 4,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 5,
    isBookmarked: true,
    isPremium: false,
    price: 132,
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    hostStatus: false,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  But where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `84578`,
    city: {
      name: `Paris`,
      coords: [53.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3809553943508, 4.939309666406198],
    title: `Nice, cozy, warm big bed apartment`,
    previewImg: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 4,
    maxAdults: 5,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Private room`,
    rating: 2,
    isBookmarked: false,
    isPremium: true,
    price: 180,
    hostName: `Max`,
    hostAvatar: `img/avatar-max.jpg`,
    hostStatus: true,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera.`
  }
];
