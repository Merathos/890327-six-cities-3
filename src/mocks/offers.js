export default [
  {
    id: `41324`,
    name: `Beautiful &amp; luxurious apartment at great location`,
    picture: `apartment-01`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 3,
    isBookmarked: true,
    isPremium: false,
    price: 120,
    hostName: `Angelina`,
    hostAvatar: `avatar-angelina`,
    hostStatus: `pro`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `73645`,
    name: `Wood and stone place`,
    picture: `apartment-02`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 2,
    maxAdults: 3,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    type: `Private room`,
    rating: 4,
    isBookmarked: false,
    isPremium: true,
    price: 80,
    hostName: `Max`,
    hostAvatar: `avatar-max`,
    hostStatus: ``,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `23455`,
    name: `Canal View Prinsengracht`,
    picture: `apartment-03`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 3,
    maxAdults: 4,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 5,
    isBookmarked: true,
    isPremium: false,
    price: 132,
    hostName: `Angelina`,
    hostAvatar: `avatar-angelina`,
    hostStatus: ``,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  But where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `84578`,
    name: `Nice, cozy, warm big bed apartment`,
    picture: `apartment-01`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 4,
    maxAdults: 5,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Private room`,
    rating: 2,
    isBookmarked: false,
    isPremium: true,
    price: 180,
    hostName: `Max`,
    hostAvatar: `avatar-max`,
    hostStatus: `pro`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera.`
  }
];
