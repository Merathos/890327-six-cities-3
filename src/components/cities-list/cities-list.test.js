import React from 'react';
import renderer from 'react-test-renderer';
import CitiesList from "./cities-list";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import offers from "../../__mocks__/offers.js";

const mockStore = configureStore([]);

const cities = [
  {
    name: `Paris`,
    coords: [4134, 123123]
  },
  {
    name: `Paris`,
    coords: [4134, 123123]
  },
  {
    name: `Paris`,
    coords: [4134, 123123]
  },
  {
    name: `Paris`,
    coords: [4134, 123123]
  },
  {
    name: `Paris`,
    coords: [4134, 123123]
  },
  {
    name: `Paris`,
    coords: [4134, 123123]
  },
];

it(`Should render Cities List`, () => {
  const store = mockStore({
    cities,
    currentCity: {
      name: `Paris`,
      coords: [51.38333, 4.9]
    },
    allOffers: offers,
    offers: offers.filter((offer) => offer.city === `Paris`)[0].offers
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <CitiesList />
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
