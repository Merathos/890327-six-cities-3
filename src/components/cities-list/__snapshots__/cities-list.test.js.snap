// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Cities List 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </li>
  </ul>
</section>
`;
